### PR TITLE
Allow VMs to run on CI and do not require parameters in runner

### PR DESCRIPTION
**Pull request by vboulineau**
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ogaca-dd — APPROVED** [review]
LGTM

**Review by pducolin — CHANGES_REQUESTED** [review]
We probably need the `set +x ... set -x` to mute possible leaks back, not convinced by the refactoring that removes the 2 more readable function to set a parameter or set a secret parameter

File: .gitlab/e2e.yml [hunk 1]
```diff
@@ -127,11 +127,16 @@ k8s-e2e-otlp-main:
   before_script:
     # Setup AWS Credentials
     - mkdir -p ~/.aws
-    - set +x
     - aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.agent-qa-profile --with-decryption --query "Parameter.Value" --out text >> ~/.aws/config
-    - set -x
     - export AWS_PROFILE=agent-qa-ci
+    # Now all `aws` commands target the agent-qa profile
+    - aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.ssh_public_key --with-decryption --query "Parameter.Value" --out text > $E2E_PUBLIC_KEY_PATH
+    - aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.ssh_key --with-decryption --query "Parameter.Value" --out text > $E2E_PRIVATE_KEY_PATH
```
[the remark this answers is not in the document]
Comment (vboulineau) [reply .gitlab/e2e.yml:134]: It's not necessary `set -x` is not default and the command does not leak information. We have usage all over the CI.

File: test/new-e2e/runner/configmap.go [hunk 2]
```diff
@@ -17,11 +17,14 @@ import (
...
-	AwsKeyPairName            = commonconfig.DDInfraConfigNamespace + ":" + infraaws.DDInfraDefaultKeyPairParamName
-	AwsPublicKeyPath          = commonconfig.DDInfraConfigNamespace + ":" + infraaws.DDinfraDefaultPublicKeyPath
+	AgentAPIKey = commonconfig.DDAgentConfigNamespace + ":" + commonconfig.DDAgentAPIKeyParamName
+	AgentAPPKey = commonconfig.DDAgentConfigNamespace + ":" + commonconfig.DDAgentAPPKeyParamName
+
 	InfraEnvironmentVariables = commonconfig.DDInfraConfigNamespace + ":" + commonconfig.DDInfraEnvironment
+
+	AWSKeyPairName    = commonconfig.DDInfraConfigNamespace + ":" + infraaws.DDInfraDefaultKeyPairParamName
+	AWSPublicKeyPath  = commonconfig.DDInfraConfigNamespace + ":" + infraaws.DDinfraDefaultPublicKeyPath
+	AWSPrivateKeyPath = commonconfig.DDInfraConfigNamespace + ":" + "aws/defaultPrivateKeyPath"
```
[the remark this answers is not in the document]
Comment (vboulineau) [reply test/new-e2e/runner/configmap.go:27]: It's private

File: test/new-e2e/runner/configmap.go [hunk 3]
```diff
@@ -43,55 +46,45 @@ func (cm ConfigMap) ToPulumi() auto.ConfigMap {
...
-		return err
-	}
-
-	cm[configMapKey] = auto.ConfigValue{
-		Value: val,
-	}
-	return nil
-}
-
-func SetConfigMapFromSecret(secretStore parameters.Store, cm ConfigMap, paramName parameters.StoreKey, configMapKey string) error {
```
[the remark this answers is not in the document]
Comment (vboulineau) [reply test/new-e2e/runner/configmap.go:58]: It's either 1 generic or 3 (generic + two public variants) functions. IMHO not worth it considering the usage of the function.

File: test/new-e2e/runner/configmap.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (pducolin) [reply test/new-e2e/runner/configmap.go:27]: For other params we first moved params to public in test-infra and then imported them here, I understand it is pedantic but it helps discoverability and readability in the end.

File: test/new-e2e/runner/configmap.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (pducolin) [reply test/new-e2e/runner/configmap.go:58]:
```go
err = SetConfigMapFromParameter(profile.SecretStore(), cm, parameters.APPKey, AgentAPPKey, true)
```

is less explicit and readable than

```go
err = SetConfigMapFromSecret(profile.SecretStore(), cm, parameters.APPKey, AgentAPPKey)
```

A reader falling on the first will have to click and follow the inner code to understand what the `true` is about, while it is explicit on the second. Not worth the refactoring to one single function.

File: .gitlab/e2e.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pducolin) [reply .gitlab/e2e.yml:134]: Asking as @KSerrania  asked for this change on the preview PR and it was applied, so I was wondering why changing it back.

**Review by pducolin — APPROVED** [review]
Thank you for addressing the comments and for allowing running tests on the CI !

**Pull request merged**